xbmc/windowing/windows/VideoSyncD3D.cpp

```diff
@@ -120,14 +164,33 @@ void CVideoSyncD3D::Cleanup()
 
 float CVideoSyncD3D::GetFps()
 {
-  DXGI_MODE_DESC DisplayMode = {};
-  DX::DeviceResources::Get()->GetDisplayMode(&DisplayMode);
+#ifdef TARGET_WINDOWS_DESKTOP
+  DEVMODEW sDevMode = {};
+  sDevMode.dmSize = sizeof(sDevMode);
+
+  if (EnumDisplaySettingsW(m_outputDesc.DeviceName, ENUM_CURRENT_SETTINGS, &sDevMode))
+  {
+    if ((sDevMode.dmDisplayFrequency + 1) % 24 == 0 || (sDevMode.dmDisplayFrequency + 1) % 30 == 0)
+      m_fps = static_cast<float>(sDevMode.dmDisplayFrequency + 1) / 1.001f;
+    else
+      m_fps = static_cast<float>(sDevMode.dmDisplayFrequency);
+
+    if (sDevMode.dmDisplayFlags & DM_INTERLACED)
+      m_fps *= 2;
+  }
+#else
+  using namespace winrt::Windows::Graphics::Display::Core;
 
-  m_fps = (DisplayMode.RefreshRate.Denominator != 0) ? (float)DisplayMode.RefreshRate.Numerator / (float)DisplayMode.RefreshRate.Denominator : 0.0f;
+  auto hdmiInfo = HdmiDisplayInformation::GetForCurrentView();
+  if (hdmiInfo) // Xbox only
+  {
+    auto currentMode = hdmiInfo.GetCurrentDisplayMode();
+    m_fps = currentMode.RefreshRate();
```

Comment:
`m_fps` is float and `currentMode.RefreshRate();` double --> needs static_cast

T:\KODI\kodi\xbmc\windowing\windows\VideoSyncD3D.cpp(188,36): warning C4244: '=': conversion from 'double' to 'float', possible loss of data